### ISSUE TITLE
The card is empty, even if the "map sensor" exists

### ISSUE THREAD

**Issue by teras** [azuwis/zigbee2mqtt-networkmap#41]
**Describe the bug**
After setting up the "map sensor" (backend setup)  and the lovelace card through HACS (frontend setup), and adding the card, the map appears still to be empty.  

**Screenshots**
This is how the component looks like
![image](https://user-images.githubusercontent.com/592568/199593870-616fb064-f019-4f49-a761-f072be0639ab.png)

**Home Assistant version**
 Home Assistant 2022.10.5 
Supervisor 2022.10.0 
Operating System 9.3 
Frontend 20221010.0 - latest 

**Zigbee2mqtt version**
Zigbee2MQTT version 1.28.0 (commit #03ba647)
zigbee-herdsman (0.14.62)

**OS and browser version**
Arch linux
Firefox 106.0.2

**Error message on browser**
No errors on console

**State attributes of `sensor.zigbee2mqtt_networkmap`**
```
links: 
- depth: 2
  linkquality: 64
  lqi: 64
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d000392729e'
    networkAddress: 32384
  sourceIeeeAddr: '0x00158d000392729e'
  sourceNwkAddr: 32384
  target:
    ieeeAddr: '0x00124b001b7976c1'
    networkAddress: 59235
  targetIeeeAddr: '0x00124b001b7976c1'
- depth: 0
  linkquality: 115
  lqi: 115
  relationship: 2
  routes: []
  source:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  sourceIeeeAddr: '0x00212effff05e273'
  sourceNwkAddr: 0
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 2
  linkquality: 139
  lqi: 139
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d0003497c98'
    networkAddress: 12101
  sourceIeeeAddr: '0x00158d0003497c98'
  sourceNwkAddr: 12101
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 2
  linkquality: 91
  lqi: 91
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d0003cd0a79'
    networkAddress: 51099
  sourceIeeeAddr: '0x00158d0003cd0a79'
  sourceNwkAddr: 51099
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 2
  linkquality: 192
  lqi: 192
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d0003cd0d7a'
    networkAddress: 12061
  sourceIeeeAddr: '0x00158d0003cd0d7a'
  sourceNwkAddr: 12061
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 2
  linkquality: 105
  lqi: 105
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d000462f60b'
    networkAddress: 4991
  sourceIeeeAddr: '0x00158d000462f60b'
  sourceNwkAddr: 4991
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 2
  linkquality: 60
  lqi: 60
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d000324110c'
    networkAddress: 3543
  sourceIeeeAddr: '0x00158d000324110c'
  sourceNwkAddr: 3543
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 2
  linkquality: 15
  lqi: 15
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x943469fffece9148'
    networkAddress: 44940
  sourceIeeeAddr: '0x943469fffece9148'
  sourceNwkAddr: 44940
  target:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  targetIeeeAddr: '0x00158d00054691f3'
- depth: 1
  linkquality: 252
  lqi: 252
  relationship: 2
  routes: []
  source:
    ieeeAddr: '0x00124b001b7976c1'
    networkAddress: 59235
  sourceIeeeAddr: '0x00124b001b7976c1'
  sourceNwkAddr: 59235
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'
- depth: 1
  linkquality: 252
  lqi: 252
  relationship: 2
  routes: []
  source:
    ieeeAddr: '0x00158d00054691f3'
    networkAddress: 47585
  sourceIeeeAddr: '0x00158d00054691f3'
  sourceNwkAddr: 47585
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'
- depth: 1
  linkquality: 252
  lqi: 252
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d00032dac5f'
    networkAddress: 623
  sourceIeeeAddr: '0x00158d00032dac5f'
  sourceNwkAddr: 623
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'
- depth: 1
  linkquality: 252
  lqi: 252
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d000349927b'
    networkAddress: 47501
  sourceIeeeAddr: '0x00158d000349927b'
  sourceNwkAddr: 47501
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'
- depth: 1
  linkquality: 249
  lqi: 249
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d000425e1c5'
    networkAddress: 435
  sourceIeeeAddr: '0x00158d000425e1c5'
  sourceNwkAddr: 435
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'
- depth: 1
  linkquality: 252
  lqi: 252
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0x00158d0003d21883'
    networkAddress: 60519
  sourceIeeeAddr: '0x00158d0003d21883'
  sourceNwkAddr: 60519
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'
- depth: 1
  linkquality: 252
  lqi: 252
  relationship: 1
  routes: []
  source:
    ieeeAddr: '0xa4c1383d2c1e3e16'
    networkAddress: 39851
  sourceIeeeAddr: '0xa4c1383d2c1e3e16'
  sourceNwkAddr: 39851
  target:
    ieeeAddr: '0x00212effff05e273'
    networkAddress: 0
  targetIeeeAddr: '0x00212effff05e273'

nodes: 
- definition:
    description: Zigbee smart switch
    model: BASICZBR3
    supports: switch (state), linkquality
    vendor: SONOFF
  failed: []
  friendlyName: mosquito_light
  ieeeAddr: '0x00124b001b7976c1'
  lastSeen: 1667420252882
  manufacturerName: SONOFF
  modelID: BASICZBR3
  networkAddress: 59235
  type: Router
- definition:
    description: Aqara D1 2 gang smart wall switch (with neutral wire)
    model: QBKG24LM
    supports: switch (state), power, action, operation_mode, linkquality
    vendor: Xiaomi
  failed: []
  friendlyName: bathroom_switch
  ieeeAddr: '0x00158d00054691f3'
  lastSeen: 1667420151857
  manufacturerName: LUMI
  modelID: lumi.switch.b2nacn02
  networkAddress: 47585
  type: Router
- definition:
    description: >-
      Aqara double key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG03LM
    supports: switch (state), device_temperature, action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: cellar_switch
  ieeeAddr: '0x00158d000349927b'
  lastSeen: 1667420390827
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral2
  networkAddress: 47501
  type: EndDevice
- definition:
    description: >-
      Aqara double key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG03LM
    supports: switch (state), device_temperature, action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: pc_switch
  ieeeAddr: '0x00158d000425e1c5'
  lastSeen: 1667420365357
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral2
  networkAddress: 435
  type: EndDevice
- definition:
    description: >-
      Aqara double key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG03LM
    supports: switch (state), device_temperature, action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: piano_switch
  ieeeAddr: '0x00158d0003497c98'
  lastSeen: 1667420358876
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral2
  networkAddress: 12101
  type: EndDevice
- definition:
    description: >-
      Aqara single key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG04LM
    supports: switch (state), action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: central_external_switch
  ieeeAddr: '0x00158d0003cd0a79'
  lastSeen: 1667420409460
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral1
  networkAddress: 51099
  type: EndDevice
- definition:
    description: >-
      Aqara double key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG03LM
    supports: switch (state), device_temperature, action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: tv_switch
  ieeeAddr: '0x00158d00032dac5f'
  lastSeen: 1667420412018
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral2
  networkAddress: 623
  type: EndDevice
- definition:
    description: >-
      Aqara single key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG04LM
    supports: switch (state), action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: corridor_switch
  ieeeAddr: '0x00158d0003d21883'
  lastSeen: 1667420392533
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral1
  networkAddress: 60519
  type: EndDevice
- definition:
    description: >-
      Aqara double key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG03LM
    supports: switch (state), device_temperature, action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: hall_switch
  ieeeAddr: '0x00158d000324110c'
  lastSeen: 1667420414005
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral2
  networkAddress: 3543
  type: EndDevice
- definition:
    description: >-
      Aqara single key wired wall switch without neutral wire. Doesn't work as a
      router and doesn't support power meter
    model: QBKG04LM
    supports: switch (state), action, operation_mode, linkquality
    vendor: Xiaomi
  friendlyName: spotlight_switch
  ieeeAddr: '0x00158d0003cd0d7a'
  lastSeen: 1667420386221
  manufacturerName: LUMI
  modelID: lumi.ctrl_neutral1
  networkAddress: 12061
  type: EndDevice
- definition:
    description: MiJia temperature & humidity sensor
    model: WSDCGQ01LM
    supports: battery, temperature, humidity, voltage, linkquality
    vendor: Xiaomi
  friendlyName: hallway
  ieeeAddr: '0x00158d000462f60b'
  lastSeen: 1667419878262
  manufacturerName: LUMI
  modelID: lumi.sens
  networkAddress: 4991
  type: EndDevice
- definition:
    description: MiJia door & window contact sensor
    model: MCCGQ01LM
    supports: battery, contact, voltage, power_outage_count, linkquality
    vendor: Xiaomi
  friendlyName: central_door_trap
  ieeeAddr: '0x00158d0003a88fd9'
  lastSeen: 1667419795783
  manufacturerName: LUMI
  modelID: lumi.sensor_magnet
  networkAddress: 44846
  type: EndDevice
- definition:
    description: MiJia door & window contact sensor
    model: MCCGQ01LM
    supports: battery, contact, voltage, power_outage_count, linkquality
    vendor: Xiaomi
  friendlyName: left_door_trap
  ieeeAddr: '0x00158d000392729e'
  lastSeen: 1667419222036
  manufacturerName: LUMI
  modelID: lumi.sensor_magnet
  networkAddress: 32384
  type: EndDevice
- definition:
    description: Door sensor
    model: TS0203
    supports: contact, battery_low, tamper, battery, voltage, linkquality
    vendor: TuYa
  friendlyName: zoi_trap
  ieeeAddr: '0xa4c1383d2c1e3e16'
  lastSeen: 1667413557160
  manufacturerName: _TZ3000_26fmupbb
  modelID: TS0203
  networkAddress: 39851
  type: EndDevice
- definition:
    description: Aqara wireless switch
    model: WXKG11LM
    supports: >-
      battery, voltage, action, device_temperature, power_outage_count,
      linkquality
    vendor: Xiaomi
  friendlyName: doorbell
  ieeeAddr: '0x00158d00020414de'
  lastSeen: 1667419067741
  manufacturerName: LUMI
  modelID: lumi.sensor_switch.aq2
  networkAddress: 627
  type: EndDevice
- definition:
    description: MiJia wireless switch
    model: WXKG01LM
    supports: battery, action, voltage, power_outage_count, linkquality
    vendor: Xiaomi
  friendlyName: '0x00158d0004024ade'
  ieeeAddr: '0x00158d0004024ade'
  lastSeen: 1667418674193
  manufacturerName: LUMI
  modelID: lumi.sensor_switch
  networkAddress: 42990
  type: EndDevice
- definition:
    description: Universal smart IR remote control
    model: UFO-R11
    supports: >-
      switch (state), learned_ir_code, ir_code_to_send, battery, voltage,
      linkquality
    vendor: Moes
  friendlyName: pc_irda
  ieeeAddr: '0x943469fffece9148'
  lastSeen: 1667417873775
  manufacturerName: _TZ3290_ot6ewjvmejq5ekhl
  modelID: TS1201
  networkAddress: 44940
  type: EndDevice
- definition: null
  failed: []
  friendlyName: Coordinator
  ieeeAddr: '0x00212effff05e273'
  lastSeen: null
  networkAddress: 0
  type: Coordinator

friendly_name: Zigbee2mqtt Networkmap
```

**Home Assistant log**
No related log here

**Additional context**
I can't think of anything more right now...

**Comment by azuwis** [comment]
With the state attribute you posted ([state.txt](https://github.com/azuwis/zigbee2mqtt-networkmap/files/9925334/state.txt), converted to JSON), I was able to generate a network map in my own setup, using this command to debug:

```
mosquitto_pub -h <ip-of-mqtt-server> -t 'zigbee2mqtt/bridge/response/networkmap' -f state.txt
```

<img width="507" alt="Screen Shot 2022-11-03 at 10 31 55" src="https://user-images.githubusercontent.com/9315/199638009-c17aaf7b-5d74-4fb5-8987-244cb54f03a1.png">

Home Assistant 2022.10.0, Fronend 20221005.0
Zigbee2mqtt 1.28.0
Firefox 106

According to the screenshot and state attribute you posted, the custom card is loaded, and the network map API of zigbee2mqtt worked fine. In this case, the custom card should work 🤔

Did the custom card always show `Refreshing...` for you?

Can you please try:

1. Reboot the machine run Home Assistant and Zigbee2mqtt
2. Wait 10 minutes, let the services settle down
3. Open Home Assistant Web UI, press custom card `REFRESH` button once
4. Wait about 2 minutes for the network map API result

**Comment by mfalkvidd** [comment]
Maybe there is a browser plugin (adblocker for example) that is blocking some of the map’s content?

**Comment by azuwis** [comment]
@teras Did you forget to put

```
        entity: sensor.zigbee2mqtt_networkmap
```

after

```
      - type: custom:zigbee2mqtt-networkmap
```
?

The full config is:

```
views:
  - title: Zigbee Network
    cards:
      - type: custom:zigbee2mqtt-networkmap
        entity: sensor.zigbee2mqtt_networkmap
```

I don't use the Web UI to edit lovelace card, you have to figure out the right syntax youself.

**Comment by teras** [comment]
I think I answered this thread and said what the problem was... but I can't find my answer here.... :open_mouth: 
Anyway, I am writing this again for future reference.
The problem was that I missed the "entity" configuration. 

I asked if I could create a MR, but I already did and was approved... than you @azuwis 

For this reason I think we can close this issue. Again thank you for the excellent support.
